File: src/main/dialogs/QRCodeDisplayer.java
package main.dialogs;

import main.interfaces.DefaultWindowsInterface;

import javax.swing.*;
import java.awt.*;

public class QRCodeDisplayer extends JDialog {

    public QRCodeDisplayer(Window parent) {
        super(parent);
        setTitle("Scan to Visit!");

        // Note: Image path might need adjustment depending on execution context, but
        // keeping as is.
        ImageIcon qrImg = new ImageIcon("ReadmeImages/GitHubVisai.png");
        Image scaledImage = qrImg.getImage().getScaledInstance(256, 256, Image.SCALE_SMOOTH);

        JLabel qrLabel = new JLabel(new ImageIcon(scaledImage));
        setResizable(false);
        qrLabel.setPreferredSize(new Dimension(256, 256));
        setSize(new Dimension(300, 300));

        // DefaultWindowsInterface.backgroundColor must be accessible
        qrLabel.setBackground(DefaultWindowsInterface.backgroundColor);
        qrLabel.setOpaque(true); // Needed for background color to show on JLabel

        add(qrLabel);
        setLocationRelativeTo(parent);
    }
}

File: src/Windows/dialogs/QRCodeDisplayer.kt
package Windows.dialogs
import Windows.interfaces.DefaultWindowsInterface
import java.awt.Dimension
import java.awt.Image
import java.awt.Window
import javax.swing.ImageIcon
import javax.swing.JDialog
import javax.swing.JLabel
class QRCodeDisplayer(parent: Window?) : JDialog(parent){
    private val qrLabel: JLabel
    init{
        title="Scan to Visit!"
        val qrImg= ImageIcon("ReadmeImages/GitHubVisai.png")
        val scaledImage=qrImg.image.getScaledInstance(256,256,Image.SCALE_SMOOTH)
        qrLabel= JLabel(ImageIcon(scaledImage))
        isResizable=false
        qrLabel.preferredSize= Dimension(256,256)
        size= Dimension(300,300)
        qrLabel.background= DefaultWindowsInterface.themeColorBG
        this.add(qrLabel)
        setLocationRelativeTo(parent)
    }
}
